metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -39,7 +38,8 @@ final class Diagnostics(
     buffers: Buffers,
     languageClient: LanguageClient,
     statistics: StatisticsConfig,
-    config: () => UserConfiguration
+    config: () => UserConfiguration,
+    workspace: AbsolutePath
```

Comment:
Not a big deal now, but just digitally thinking out loud here. I know we were thinking about trying to avoid having to pass the workspace all around, which delays the creation of some classes. I see we use it in here only to see if a file is a read only file. If you see an easy way to do this without including the workspace, it may be preferable. If not, no worries.

Reply:
Changed it to option, so it should work fine in both case.